### ISSUE TITLE
Issue with tagged enum

### ISSUE THREAD

**Issue by kodieg** [mockersf/hocon.rs#21]
Parsing of tagged enum doesn't seem to work properly. 

I have following definitions:
```rust
#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash)]
#[serde(rename_all = "camelCase")]
pub struct RequestOptions {
    #[serde(default)]
    #[serde(with = "humantime_serde")]
    timeout:      Option<Duration>,
    retry_policy: Option<RetryPolicy>,
}

#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash)]
#[serde(tag = "type")]
#[serde(rename_all = "camelCase")]
pub enum RetryPolicy {
    None,
    #[serde(rename = "camelCase")]
    Asap {
        num_retries: u32,
    },
}
```

Parsing sample input does not recognizes variant:

```rust
let s = r#"
retryPolicy {
     type: asap
     numRetries: 2
 }
"#;
let config: RequestOptions = HoconLoader::new().load_str(s).unwrap().resolve().unwrap();
```

Above code panics.

**Comment by mockersf** [comment]
Thank you for the bug report ! It should be fixed in v0.3.3 released just now

by the way, there is an issue in your serde attribute, it should be:
```rust
#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash)]
#[serde(rename_all = "camelCase")]
pub struct RequestOptions {
    #[serde(default)]
    #[serde(with = "humantime_serde")]
    timeout:      Option<Duration>,
    retry_policy: Option<RetryPolicy>,
}

#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash)]
#[serde(tag = "type")]
#[serde(rename_all = "camelCase")]
pub enum RetryPolicy {
    None,
    #[serde(rename_all = "camelCase")]
    Asap {
        num_retries: u32,
    },
}
```

**Comment by kodieg** [comment]
Thank you!
